lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -4629,8 +4629,14 @@ ConstraintResult GenericSignatureBuilder::addTypeRequirement(
                         ->getDependentType(getGenericParams());
 
       Impl->HadAnyError = true;
-      Diags.diagnose(source.getLoc(), diag::requires_conformance_nonprotocol,
-                     subjectType, constraintType);
+      
+      if (subjectType->is<DependentMemberType>()) {
+        subjectType = resolveDependentMemberTypes(*this, subjectType);
```

Comment:
What is this for?

Reply:
@slavapestov This is because getting the requirement source triggers an assert for

```swift
protocol D {
  associatedtype Foo where Foo: String
}
```

in `DependentMemberType::get`